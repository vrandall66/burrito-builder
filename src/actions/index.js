export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const addOrder = order => ({
  type: 'ADD_ORDER',
  order
});
