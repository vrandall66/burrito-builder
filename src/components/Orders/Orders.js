import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setOrders } from '../../actions';
import { getOrders } from '../../apiCalls';
import './Orders.css';

export class Orders extends Component {
  componentDidMount() {
    const { setOrders } = this.props;
    getOrders()
      .then(data => setOrders(data.orders))
      .catch(err => console.error('Error fetching:', err));
  }

  mapThroughOrders = () => {
    const { orders } = this.props;
    const orderEls = orders.map(order => {
      return (
        <div className='order'>
          <h3>{order.name}</h3>
          <ul className='ingredient-list'>
            {order.ingredients.map(ingredient => {
              return <li>{ingredient}</li>;
            })}
          </ul>
        </div>
      );
    });
    return orderEls;
  };

  render() {
    let orderEls = this.mapThroughOrders();
    return (
      <section>{orderEls.length ? orderEls : <p>No orders yet!</p>}</section>
    );
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => {
  bindActionCreators(
    {
      setOrders
    },
    dispatch
  );
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Orders);
