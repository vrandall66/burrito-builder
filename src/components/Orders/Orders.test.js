import React from 'react';
import { shallow } from 'enzyme';
import { setOrders } from '../../actions';
import { Orders, mapStateToProps, mapDispatchToProps } from './Orders';

describe('Orders', () => {
  let wrapper;
  let orders = [
    { ingredients: [{}, {}] },
    { ingredients: [{}, {}] },
    { ingredients: [{}, {}] }
  ];

  beforeEach(() => {
    wrapper = shallow(<Orders orders={orders} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an object with an orders object', () => {
    const mockStoreState = {
      orders: { orders: [{}, {}, {}] }
    };

    const expected = {
      orders: { orders: [{}, {}, {}] }
    };

    const mappedProps = mapStateToProps(mockStoreState);

    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('calls dispatch with setOrders', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setOrders([{}, {}, {}]);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setOrders([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
