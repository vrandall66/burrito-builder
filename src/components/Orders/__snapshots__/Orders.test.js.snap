// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="0"
  >
    <h3 />
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        <Component />
      </li>
      <li
        key="1"
      >
        <Component />
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="1"
  >
    <h3 />
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        <Component />
      </li>
      <li
        key="1"
      >
        <Component />
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="2"
  >
    <h3 />
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        <Component />
      </li>
      <li
        key="1"
      >
        <Component />
      </li>
    </ul>
  </div>
</section>
`;
